spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/security/servlet/RequestMatcherProvider.java

```diff
@@ -33,6 +35,17 @@
 	 * @param pattern the request pattern
 	 * @return a request matcher
 	 */
-	RequestMatcher getRequestMatcher(String pattern);
+	default RequestMatcher getRequestMatcher(String pattern) {
+		return getRequestMatcher(pattern, null);
+	}
+
+	/**
+	 * Return the {@link RequestMatcher} to be used for the specified pattern and http
+	 * method.
+	 * @param pattern the request pattern
+	 * @param httpMethod the http method
+	 * @return a request matcher
+	 */
+	RequestMatcher getRequestMatcher(String pattern, HttpMethod httpMethod);
```

Comment:
I am not super-excited about this change. However, this seemed the least intrusive from a breakage standpoint. This leaves the other API in tact but it would break any previous lambda usage as the functional interface method now takes `(pattern, method)` vs. the previous `(pattern)`.

Reply:
I've been playing with this locally and I'll try to push something soon. I think this API is really in wrong place so we could deprecate it here and move it to `org.springframework.boot.actuate.autoconfigure.security.servlet` where it's closer to the actual thing that uses it.